src/pages/Registration.tsx

```diff
@@ -0,0 +1,117 @@
+                                            <InputRightAddon><Icon width='1.45rem' as={EmailIcon}/></InputRightAddon>
+                                        </InputGroup>
+                                        <FormErrorMessage justifyContent='end'>{fieldProps.meta.value && fieldProps.meta.error}</FormErrorMessage>
+                                    </FormControl>} />
+                                <Field name='password' children={(fieldProps: FieldProps) =>
+                                    <FormControl isInvalid={fieldProps.meta.value && fieldProps.meta.error}>
+                                        <FormLabel color='blue.700'><strong>Password</strong></FormLabel>
+                                        <InputGroup>
+                                            <Input placeholder='Password' {...fieldProps.field} minLength={6} maxLength={20} type={show ? 'text' : 'password'}/>
+                                            <InputRightAddon width='4.5rem'>
+                                                <Button h='1.75rem' size='sm' onClick={handleClick}>
+                                                    {show ? 'Hide' : 'Show'}
+                                                </Button>
+                                            </InputRightAddon>
+                                        </InputGroup>
+                                        <FormErrorMessage justifyContent='end'>{fieldProps.meta.value && fieldProps.meta.error}</FormErrorMessage>
+                                    </FormControl>} />
+                                <Field name='repeatPassword' children={(fieldProps: FieldProps) =>
+                                    <FormControl isInvalid={fieldProps.meta.value && fieldProps.meta.error}>
+                                        <FormLabel color='blue.700'><strong>Repeat Password</strong></FormLabel>
+                                        <InputGroup>
+                                            <Input placeholder='Repeat Password' {...fieldProps.field} minLength={6} maxLength={20} type={show ? 'text' : 'password'}/>
+                                            <InputRightAddon width='4.5rem'>
+                                                <Button h='1.75rem' size='sm' onClick={handleClick}>
+                                                    {show ? 'Hide' : 'Show'}
+                                                </Button>
+                                            </InputRightAddon>
+                                        </InputGroup>
+                                        <FormErrorMessage justifyContent='end'>{fieldProps.meta.value && fieldProps.meta.error}</FormErrorMessage>
+                                    </FormControl>} />
+                            </VStack>
+                            <Button colorScheme='blue' width={200} type='submit'
+                                    disabled={!formProps.values.email || !formProps.values.password ||
```

Comment:
Since we added yup validation we can generally check if the values are valid via formProps.errors.[field name], and this will return a falsy value if there are no errors  for the field or it will return the error message (as we defined in the yup schema) if there is any error.
Specifically here we can do:
disabled={formProps.errors.email || formProps.errors.password}
or to check all fields at once:
disabled={!!formProps.errors}